src/models/budget.model.ts

```diff
@@ -0,0 +1,24 @@
+import { model, Schema, Types } from "mongoose";
+import { IBudget } from "../types/budget";
+
+const budgetSchema = new Schema<IBudget>({
+  user: { type: Schema.Types.ObjectId, required: true, ref: "User" },
+  categories: {
+    type: [{ type: Schema.Types.ObjectId, ref: "Category", required: true }],
+    validate: [
+      (val: Array<Types.ObjectId>) => val.length > 0 && val.length <= 12,
+      "Only 1 to 12 categories are allowed",
+    ],
+  },
+  month: {
+    type: Number,
+    required: [true, "Please add month"],
+    min: 1,
```

Comment:
We're using `dayjs` as the date management lib in the frontend. 

According to **[dayjs documentation](https://day.js.org/docs/en/get-set/month)**, it might be easier to keep the month range from `0` to `11`